File: backup/app/schemas/graph.py

# File: app/schemas/graph.py
from pydantic import BaseModel
from typing import List, Dict, Any, Optional
from datetime import datetime
from .common import GraphStatus

class NodeBase(BaseModel):
    type: str
    position: Dict[str, float]
    data: Dict[str, Any]

class EdgeBase(BaseModel):
    source: str
    target: str
    type: str

class GraphCreate(BaseModel):
    name: str
    nodes: List[NodeBase]
    edges: List[EdgeBase]

class Graph(GraphCreate):
    id: int
    status: GraphStatus
    created_at: datetime
    updated_at: datetime
    owner_id: int

    class Config:
        from_attributes = True

File: backup/src/types/graph.ts
// File: src/types/graph.ts
export interface Graph {
    id: string;
    name: string;
    nodes: Node[];
    edges: Edge[];
    status: 'draft' | 'ready' | 'deployed' | 'error';
    createdAt: string;
    updatedAt: string;
}

interface Node {
    id: string;
    type: string;
    position: { x: number; y: number };
    data: Record<string, any>;
}

interface Edge {
    id: string;
    source: string;
    target: string;
    type: string;
}
